stdlib/public/Platform/POSIXError.swift

```diff
@@ -554,4 +552,130 @@ public enum POSIXErrorCode : Int32 {
   case ENOTRECOVERABLE = 131
 }
 
+#elseif os(Windows)
+
+/// Enumeration describing POSIX error codes.
+public enum POSIXErrorCode : Int32 {
```

Comment:
Hang on, if they're actually POSIX error codes, shouldn't there only be one enum?

Reply:
No, even between Darwin and Linux the same definitions have different underlying values.
e.g. `EDEADLK`
- Darwin: https://github.com/apple/swift/blob/32fb4980e8b30a8e354eb98607d8c49069b6224b/stdlib/public/Platform/POSIXError.swift#L39
- Linux: https://github.com/apple/swift/blob/32fb4980e8b30a8e354eb98607d8c49069b6224b/stdlib/public/Platform/POSIXError.swift#L341
- Windows: https://github.com/apple/swift/blob/32fb4980e8b30a8e354eb98607d8c49069b6224b/stdlib/public/Platform/POSIXError.swift#L657

Not to mentions the fact that Darwin has a lot of error codes like `ENOPOLICY` that don't even exist in other OS.

Reply:
This inconsistency has already caused issues for me on Linux before https://github.com/apple/swift/pull/24028

Reply:
@jrose-apple - the only thing that it means for them to be POSIX error codes it that they have similar names, the values are implementation defined :-D (Isn't POSIX wonderful?)

Reply:
Hm, okay. Thanks for clearing that up.

Reply:
Please see page 220 of the POSIX.1-2001 spec:
http://www.open-std.org/jtc1/sc22/open/n4217.pdf

It defines some macros, and that `errno` is an ISO C `int`, that is all.

In fact, since `errno` is just a C int with some macros the existence of `POSIXErrorCode` seems questionable to me, but I guess every different POSIX platform can conform to the same protocol...

Reply:
@TaborKelly I also think that POSIXErrorCode could have been in the beginning an `RawRepresentable` `struct` that takes any `Int32` value instead of an enum, but since Apple made this design decision for macOS, and you get exhaustive switch statements, not to mention that it is good (easier) for the developer if this is a stable enum that doesn't change, it makes sense to imitate it for other platforms since Foundation relies on this.

Reply:
Also exhaustive switch statements are great, but this file is going to have to grow to support every OS Swift compiles for, unfortunately.

Reply:
The switch statements won't be exhaustive because POSIXError is in a library built with library evolution support, but we do still have all the other worries, sure.

Reply:
Well, relatively stable, I doubt the next versions of macOS is going to introduce new error codes.